intel-sgx/scripts/gdb.py

```diff
@@ -30,8 +38,35 @@ def read_long(inferior, address, length):
     ret += b
   return ret
 
-def find_vma_base(addr):
-  s = gdb.execute("info proc mappings", False, True)
+def find_vma_base_from_runner(addr):
+  """Given an address `addr`, find the baseaddress of the enclave that contains the `addr`.
+
+  This function uses a feature of the new ftxsgx-runners (2023-08),
+  and works on both the Montgomery loader and the Augusta loader.
+  """
+  lang_restore = gdb.parameter('language')
+  # Old versions of GDB such as gdb-9.2 on Ubuntu-20.04 do not have `gdb.with_parameter`.
+  # We therefore have to override the language first,
+  # and then carefully restore the old value after.
+  gdb.execute("set language c", from_tty = False, to_string = True)
+
+  # Example output: "$3 = 140737219919872"
+  func_result = gdb.execute(
+    "print (unsigned long) ftxsgx_get_baseaddress_from_pointer({})".format(addr),
+    from_tty = False,
+    to_string = True
+  )
```

Comment:
On an old runner, this still gives me:
```
raoul@sacagawea:/tmp/app/target/x86_64-fortanix-unknown-sgx/debug$ gdb --args ftxsgx-runner app.sgxs
GNU gdb (Ubuntu 9.2-0ubuntu1~20.04.1) 9.2
Copyright (C) 2020 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
Type "show copying" and "show warranty" for details.
This GDB was configured as "x86_64-linux-gnu".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<http://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
    <http://www.gnu.org/software/gdb/documentation/>.

For help, type "help".
Type "apropos word" to search for commands related to "word"...
Reading symbols from ftxsgx-runner...
(gdb) source /tmp/gdb.py 
(gdb) r
Starting program: /home/raoul/.cargo/bin/ftxsgx-runner app.sgxs
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".
[New Thread 0x7ffff78a1700 (LWP 159330)]
[New Thread 0x7ffff76a0700 (LWP 159331)]
[New Thread 0x7ffff749f700 (LWP 159332)]
[New Thread 0x7ffff729e700 (LWP 159333)]
[New Thread 0x7ffff709d700 (LWP 159334)]
[New Thread 0x7ffff6e9c700 (LWP 159335)]
[New Thread 0x7ffff6c9b700 (LWP 159336)]
[New Thread 0x7ffff6a9a700 (LWP 159337)]
Attaching debugger

Thread 1 "ftxsgx-runner" received signal SIGTRAP, Trace/breakpoint trap.
0x00005555555a9891 in enclave_runner::usercalls::EnclaveState::handle_usercall::{{closure}} ()
(gdb) sgxstate auto app
Python Exception <class 'gdb.error'> No symbol "ftxsgx_get_baseaddress_from_pointer" in current context.: 
Error occurred in Python: No symbol "ftxsgx_get_baseaddress_from_pointer" in current context.
```
Note that the runner indeed did not have the `ftxsgx_get_baseaddress_from_pointer` function but it shouldn't crash and burn like that. Can you try adding a try catch block?

Reply:
Good catch! (I should've seen that coming myself as well.)
Since this PR intends to cover https://github.com/fortanix/rust-sgx/issues/490, I'll definitely fix it first

Reply:
Fixed